### ISSUE TITLE
`redudant-import-alias` false positive

### ISSUE THREAD

**Issue by denisvmedia** [mgechev/revive#1662]
**Describe the bug**

Linter reports it as:

```
test.go:19:2: redundant-import-alias: Import alias "utls" is redundant (revive)
	utls "github.com/enetx/utls"
```

But the fact is that the package is actually named tls, and thus aliasing to utls is not redundant (and is preferred if `"crypto/tls"` at the same time).

**To Reproduce**

Try importing a package named boo, but having a package uri as `github.com/orgname/foo.

**Expected behavior**
No reporting.

**Desktop (please complete the following information):**
 - OS: macOS Tahoe 26.3
 - go 1.26.0
 - revive used through golangci-lint 1.10.1

**Additional context**
I have a feeling we already discussed this issue, but I might be mistaken.

**Comment by alexandear** [comment]
I believe we have a similar issue #936, and the answer https://github.com/mgechev/revive/issues/936#issuecomment-2330647279

**Comment by denisvmedia** [comment]
Indeed, that's why I had de ja vu. Thanks for finding it. ~I'll close the issue then (we can't easily fix it).~ EDIT: re-open due to this comment: https://github.com/mgechev/revive/issues/936#issuecomment-3998202816 (@ccoVeille)

**Comment by ccoVeille** [comment]
Maybe I gave you hope and it won't change but a small investigation might be interesting

**Comment by alexandear** [comment]
`x/tools` v0.43.0 exposes the [`Module.GoMod`](https://pkg.go.dev/golang.org/x/tools/go/analysis#Module) field (upgraded in #1684). So now we can try fixing this false positive.

**Comment by denisvmedia** [comment]
@alexandear do you want to take it? I'm open to do it as well, just don't have time right now (but don't see an issue doing it a little bit later this month).
